src/pj_obs_api.c

```diff
@@ -296,10 +315,44 @@ size_t proj_transform (
         *z = coord.xyzt.z;
     if (nt==1)
         *t = coord.xyzt.t;
-    
+
     return i;
 }
 
+/*****************************************************************************/
+int proj_transform_obs (PJ *P, enum proj_direction direction, size_t n, PJ_OBS *obs) {
+/******************************************************************************
+    Batch transform an array of PJ_OBS.
+
+    Returns 0 if all observations are transformed without error, otherwise
+    returns error number.
```

Comment:
> Nowadays, wouldn't returning NaN be a better indication of a problem?

I think so, yes. What is the reason this hasn't been done in the past? Some compatibility thing? I was just going with the known behaviour of `pj_transform`, but I'm fine with using NaN's instead. It might require thorough run-down of the existing code to make sure that it behaves in the same way throughout the code base.